core/tvm_executor.go

```diff
@@ -28,432 +27,432 @@ import (
-				return
-			}
-			snapshot := accountdb.Snapshot()
-			if MinerManagerImpl.activateAndAddStakeMiner(miner, accountdb, height) &&
-				MinerManagerImpl.AddStakeDetail(miner.ID, miner, miner.Stake, accountdb) {
-				accountdb.SubBalance(*transaction.Source, amount)
-				Logger.Debugf("TVMExecutor Execute MinerApply success(activate) Source %s", transaction.Source.Hex())
-				success = true
-			} else {
-				accountdb.RevertToSnapshot(snapshot)
-			}
-		} else {
-			Logger.Debugf("TVMExecutor Execute MinerApply Fail(Already Exist) Source %s", transaction.Source.Hex())
-		}
-		return
+func (ss *rewardExecutor) ParseTransaction() error {
+	rm := BlockChainImpl.GetRewardManager()
+
+	_, targets, blockHash, packFee, err := rm.ParseRewardTransaction(ss.tx)
+	if err != nil {
+		return err
 	}
-	miner.ApplyHeight = height
-	miner.Status = types.MinerStatusNormal
-	if MinerManagerImpl.addMiner(transaction.Source[:], miner, accountdb) > 0 &&
-		MinerManagerImpl.AddStakeDetail(miner.ID, miner, miner.Stake, accountdb) {
-		accountdb.SubBalance(*transaction.Source, amount)
-		Logger.Debugf("TVMExecutor Execute MinerApply Success Source:%s Height:%d", transaction.Source.Hex(), height)
-		success = true
+
+	ss.blockHash = blockHash
+
+	// Reward for each target address
+	ss.reward = ss.tx.Value.Value()
+	ss.packFee = packFee
+
+	ss.targets = make([]common.Address, 0)
+	for _, tid := range targets {
+		ss.targets = append(ss.targets, common.BytesToAddress(tid))
 	}
 
-	return success, err, cumulativeGasUsed
+	// Check if the corresponding block exists
+	if bh := BlockChainImpl.QueryBlockHeaderByHash(ss.blockHash); bh == nil {
+		return fmt.Errorf("block not exist：%v", ss.blockHash.Hex())
+	} else {
+		ss.blockHeight = bh.Height
+	}
+	// Check if there is a reward transaction of the same block already executed
+	if rm.contain(ss.blockHash.Bytes(), ss.accountDB) {
+		return fmt.Errorf("reward transaction already executed:%v", ss.blockHash.Hex())
+	}
+	ss.proposal = common.BytesToAddress(ss.bh.Castor)
+	return nil
 }
 
-func (executor *TVMExecutor) executeMinerStakeTx(accountdb *account.AccountDB, transaction *types.Transaction, height uint64, castor common.Address, gasUsed *types.BigInt) (success bool, err *types.TransactionError, cumulativeGasUsed uint64) {
-	Logger.Debugf("Execute miner Stake tx:%s,source: %v\n", transaction.Hash.Hex(), transaction.Source.Hex())
-	success = false
+func (ss *rewardExecutor) Transition() *result {
+	ret := newResult()
+	// Add the balance of the target addresses for verifying the block
+	// Including the verifying reward and gas fee share
+	for _, addr := range ss.targets {
+		ss.accountDB.AddBalance(addr, ss.reward)
+	}
+	ret.cumulativeGas = new(big.Int)
 
-	var _type, id, value = MinerManagerImpl.Transaction2MinerParams(transaction)
-	amount := new(big.Int).SetUint64(value)
+	// Add the balance of proposer with pack fee for packing the reward tx
+	ss.accountDB.AddBalance(ss.proposal, ss.packFee)
```

Comment:
why not treat the packFee as cumulativeGas so that is can be display in the receipt

Reply:
reward for the block